22:25:21 [ERROR] /root/starrocks/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Analyzer.java:[415,51] cannot find symbol
22:25:21   symbol:   class ShowRoutineLoadStmt
22:25:21   location: class com.starrocks.sql.analyzer.Analyzer.AnalyzerVisitor
22:25:21 [INFO] 1 error
22:25:21 [INFO] -----------

@shshenhua Hi, the FE compiled failed.

--- a/fe/fe-core/src/main/java/com/starrocks/analysis/PauseRoutineLoadStmt.java
+++ b/fe/fe-core/src/main/java/com/starrocks/analysis/PauseRoutineLoadStmt.java
@@ -1,24 +1,11 @@
-// Licensed to the Apache Software Foundation (ASF) under one
-// or more contributor license agreements.  See the NOTICE file
-// distributed with this work for additional information
-// regarding copyright ownership.  The ASF licenses this file
-// to you under the Apache License, Version 2.0 (the
-// "License"); you may not use this file except in compliance
-// with the License.  You may obtain a copy of the License at
-//
-//   http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing,
-// software distributed under the License is distributed on an
-// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
-// KIND, either express or implied.  See the License for the
-// specific language governing permissions and limitations
-// under the License.
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
 
 package com.starrocks.analysis;
 
+import com.starrocks.cluster.ClusterNamespace;
 import com.starrocks.common.AnalysisException;
 import com.starrocks.common.UserException;
+import com.starrocks.sql.ast.AstVisitor;
 
 /*
   Pause routine load by name
@@ -28,7 +15,7 @@ import com.starrocks.common.UserException;
  */
 public class PauseRoutineLoadStmt extends DdlStmt {
 
-    private final LabelName labelName;
+    private LabelName labelName;
 
     public PauseRoutineLoadStmt(LabelName labelName) {
         this.labelName = labelName;
@@ -42,9 +29,32 @@ public class PauseRoutineLoadStmt extends DdlStmt {
         return labelName.getDbName();
     }
 
+    public void setLabelName(LabelName labelName) {
+        this.labelName = labelName;
+    }
+
+    @Deprecated
     @Override
     public void analyze(Analyzer analyzer) throws AnalysisException, UserException {
         super.analyze(analyzer);
         labelName.analyze(analyzer);
+    }
+
+    @Override
+    public String toSql() {
+        String dbName = labelName.getDbName();
+        String routineName = labelName.getLabelName();
+        dbName = ClusterNamespace.getNameFromFullName(dbName);
+        return "PAUSE ROUTINE LOAD FOR " + dbName + "." + routineName;
+    }
+
+    @Override
+    public <R, C> R accept(AstVisitor<R, C> visitor, C context) {
+        return visitor.visitPauseRoutineLoadStatement(this, context);
+    }
+
+    @Override
+    public boolean isSupportNewPlanner() {
+        return true;
     }
 }

--- a/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Analyzer.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Analyzer.java
@@ -26,6 +26,7 @@ import com.starrocks.analysis.DropTableStmt;
 import com.starrocks.analysis.InsertStmt;
 import com.starrocks.analysis.LimitElement;
 import com.starrocks.analysis.LoadStmt;
+import com.starrocks.analysis.PauseRoutineLoadStmt;
 import com.starrocks.analysis.RecoverDbStmt;
 import com.starrocks.analysis.RecoverPartitionStmt;
 import com.starrocks.analysis.RecoverTableStmt;
@@ -389,6 +390,12 @@ public class Analyzer {
         @Override
         public Void visitRecoverPartitionStmt(RecoverPartitionStmt statement, ConnectContext context) {
             RecoverPartitionAnalyzer.analyze(statement, context);
+            return null;
+        }
+
+        @Override
+        public Void visitPauseRoutineLoadStatement(PauseRoutineLoadStmt statement, ConnectContext session) {
+            PauseRoutineLoadAnalyzer.analyze(statement, session);
             return null;
         }
 

--- a/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PauseRoutineLoadAnalyzer.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PauseRoutineLoadAnalyzer.java
@@ -1,0 +1,32 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+package com.starrocks.sql.analyzer;
+
+import com.google.common.base.Strings;
+import com.starrocks.analysis.LabelName;
+import com.starrocks.analysis.PauseRoutineLoadStmt;
+import com.starrocks.cluster.ClusterNamespace;
+import com.starrocks.common.ErrorCode;
+import com.starrocks.common.ErrorReport;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.server.CatalogMgr;
+
+public class PauseRoutineLoadAnalyzer {
+    public static void analyze(PauseRoutineLoadStmt statement, ConnectContext context) {
+        String db = statement.getDbFullName();
+        String catalog = context.getCurrentCatalog();
+        if (Strings.isNullOrEmpty(db)) {
+            db = context.getDatabase();
+            if (CatalogMgr.isInternalCatalog(catalog)) {
+                db = ClusterNamespace.getFullName(db);
+            }
+            if (Strings.isNullOrEmpty(db)) {
+                ErrorReport.reportSemanticException(ErrorCode.ERR_NO_DB_ERROR);
+            }
+        } else {
+            if (CatalogMgr.isInternalCatalog(catalog)) {
+                db = ClusterNamespace.getFullName(db);
+            }
+        }
+        statement.setLabelName(new LabelName(db,  statement.getName()));
+    }
+}

--- a/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeChecker.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/analyzer/PrivilegeChecker.java
@@ -938,6 +938,15 @@ public class PrivilegeChecker {
             return null;
         }
 
+        @Override
+        public Void visitShowRoutineLoadStatement(ShowRoutineLoadStmt statement, ConnectContext session) {
+            String db = statement.getDbFullName();
+            if (!GlobalStateMgr.getCurrentState().getAuth().checkDbPriv(session, db, PrivPredicate.SHOW)) {
+                ErrorReport.reportSemanticException(ErrorCode.ERR_DB_ACCESS_DENIED, session.getQualifiedUser(), db);
+            }
+            return null;
+        }
+
         public Void visitShowBrokerStmt(ShowBrokerStmt statement, ConnectContext context) {
             if (!GlobalStateMgr.getCurrentState().getAuth().checkGlobalPriv(ConnectContext.get(), PrivPredicate.ADMIN)
                     && !GlobalStateMgr.getCurrentState().getAuth().checkGlobalPriv(ConnectContext.get(),

--- a/fe/fe-core/src/main/java/com/starrocks/sql/ast/AstVisitor.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/ast/AstVisitor.java
@@ -72,6 +72,7 @@ import com.starrocks.analysis.ModifyPartitionClause;
 import com.starrocks.analysis.ModifyTablePropertiesClause;
 import com.starrocks.analysis.OrderByElement;
 import com.starrocks.analysis.ParseNode;
+import com.starrocks.analysis.PauseRoutineLoadStmt;
 import com.starrocks.analysis.RecoverDbStmt;
 import com.starrocks.analysis.RecoverPartitionStmt;
 import com.starrocks.analysis.RecoverTableStmt;
@@ -358,6 +359,10 @@ public abstract class AstVisitor<R, C> {
 
     public R visitShowCreateDbStatement(ShowCreateDbStmt statement, C context) {
         return visitShowStatement(statement, context);
+    }
+
+    public R visitPauseRoutineLoadStatement(PauseRoutineLoadStmt statement, C context) {
+        return visitStatement(statement, context);
     }
 
     public R visitShowRoutineLoadStatement(ShowRoutineLoadStmt statement, C context) {

--- a/fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java
@@ -116,6 +116,7 @@ import com.starrocks.analysis.PartitionDesc;
 import com.starrocks.analysis.PartitionKeyDesc;
 import com.starrocks.analysis.PartitionNames;
 import com.starrocks.analysis.PartitionValue;
+import com.starrocks.analysis.PauseRoutineLoadStmt;
 import com.starrocks.analysis.Predicate;
 import com.starrocks.analysis.RangePartitionDesc;
 import com.starrocks.analysis.RecoverDbStmt;
@@ -1414,6 +1415,19 @@ public class AstBuilder extends StarRocksBaseVisitor<ParseNode> {
         return ret;
     }
     // ------------------------------------------- Routine Statement ---------------------------------------------------
+
+    @Override
+    public ParseNode visitPauseRoutineLoadStatement(StarRocksParser.PauseRoutineLoadStatementContext context) {
+        String database = null;
+        if (context.db != null) {
+            database = context.db.getText();
+        }
+        String name = null;
+        if (context.name != null) {
+            name = context.name.getText();
+        }
+        return new PauseRoutineLoadStmt(new LabelName(database, name));
+    }
 
     @Override
     public ParseNode visitShowRoutineLoadStatement(StarRocksParser.ShowRoutineLoadStatementContext context) {

--- a/fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4
@@ -77,6 +77,7 @@ statement
     | deleteStatement                                                                       #delete
 
     //Routine Statement
+    | pauseRoutineLoadStatement                                                             #pauseRoutineLoad
     | showRoutineLoadStatement                                                              #showRoutineLoad
 
     // Admin Statement
@@ -572,6 +573,10 @@ deleteStatement
     ;
 
 // ------------------------------------------- Routine Statement -----------------------------------------------------------
+
+pauseRoutineLoadStatement
+    : PAUSE ROUTINE LOAD FOR (db=qualifiedName '.')? name=identifier
+    ;
 
 showRoutineLoadStatement
     : SHOW ALL? ROUTINE LOAD (FOR (db=qualifiedName '.')? name=identifier)?

--- a/fe/fe-core/src/test/java/com/starrocks/analysis/PauseRoutineLoadStmtTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/analysis/PauseRoutineLoadStmtTest.java
@@ -1,0 +1,33 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.analysis;
+
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.utframe.UtFrameUtils;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+
+public class PauseRoutineLoadStmtTest {
+
+    private ConnectContext ctx;
+
+    @Before
+    public void setUp() throws Exception {
+    }
+
+    @Test
+    public void testNormal() throws Exception {
+        ctx = UtFrameUtils.createDefaultCtx();
+        ctx.setCluster("default_cluster");
+        ctx.setDatabase("testDb");
+
+        PauseRoutineLoadStmt stmt = new PauseRoutineLoadStmt(new LabelName("testDb","label"));
+
+        com.starrocks.sql.analyzer.Analyzer.analyze(stmt, ctx);
+        Assert.assertEquals("PAUSE ROUTINE LOAD FOR testDb.label", stmt.toSql());
+        Assert.assertEquals("default_cluster:testDb", stmt.getDbFullName());
+    }
+
+}

--- a/fe/fe-core/src/test/java/com/starrocks/analysis/ShowRoutineLoadStmtTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/analysis/ShowRoutineLoadStmtTest.java
@@ -39,14 +39,13 @@ public class ShowRoutineLoadStmtTest {
     @Test
     public void testNormal() throws Exception {
         ctx = UtFrameUtils.createDefaultCtx();
-        ctx.setCluster("default_cluster");
         ctx.setDatabase("testDb");
 
         ShowRoutineLoadStmt stmt = new ShowRoutineLoadStmt(new LabelName("testDb","label"), false);
 
         com.starrocks.sql.analyzer.Analyzer.analyze(stmt, ctx);
         Assert.assertEquals("SHOW ROUTINE LOAD FOR `testDb`.`label`", stmt.toString());
-        Assert.assertEquals("default_cluster:testDb", stmt.getDbFullName());
+        Assert.assertEquals("testDb", stmt.getDbFullName());
         Assert.assertFalse(stmt.isIncludeHistory());
         Assert.assertEquals(18, stmt.getMetaData().getColumnCount());
         Assert.assertEquals("Id", stmt.getMetaData().getColumn(0).getName());

